src/CachePlugin.php

```diff
@@ -67,12 +68,24 @@ public function handleRequest(RequestInterface $request, callable $next, callabl
 
         if ($cacheItem->isHit()) {
             // return cached response
-            return new FulfilledPromise($cacheItem->get());
+            $data = $cacheItem->get();
+            $response = $data['response'];
+            $response = $response->withBody($this->streamFactory->createStream($data['body']));
+
+            return new FulfilledPromise($response);
         }
 
         return $next($request)->then(function (ResponseInterface $response) use ($cacheItem) {
             if ($this->isCacheable($response)) {
-                $cacheItem->set($response)
+                $bodyStream = $response->getBody();
+                $body = $bodyStream->__toString();
+                if ($bodyStream->isSeekable()) {
+                    $bodyStream->rewind();
+                } else {
+                    $response = $response->withBody($this->streamFactory->createStream($body));
+                }
+
+                $cacheItem->set(['response' => $response, 'body' => $body])
```

Comment:
The cache implementation have no problem storing a stream object. But the stream object may wrap php//temp which wont be available when the cached object is retrieved. 

So yes, the content of the stream is ignored by the cache if we try to store a stream object.